extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/tracing/MongoTracingCommandListener.java

```diff
@@ -0,0 +1,92 @@
+package io.quarkus.mongodb.tracing;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import javax.annotation.Nullable;
+
+import jakarta.inject.Inject;
+
+import org.jboss.logging.Logger;
+
+import com.mongodb.event.*;
+
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.instrumentation.api.instrumenter.AttributesExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
+import io.opentelemetry.instrumentation.api.instrumenter.SpanKindExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.SpanNameExtractor;
+
+public class MongoTracingCommandListener implements CommandListener {
+    private static final org.jboss.logging.Logger LOGGER = Logger.getLogger(MongoTracingCommandListener.class);
+    private static final String KEY = "mongodb.command";
+    private final Map<Integer, ContextEvent> requestMap;
+    private final Instrumenter<CommandStartedEvent, Void> instrumenter;
+
+    private record ContextEvent(Context context, CommandStartedEvent commandEvent) {
+    }
+
+    @Inject
+    public MongoTracingCommandListener(OpenTelemetry openTelemetry) {
+        requestMap = new ConcurrentHashMap<>();
+        SpanNameExtractor<CommandStartedEvent> spanNameExtractor = CommandEvent::getCommandName;
+        instrumenter = Instrumenter.<CommandStartedEvent, Void> builder(
+                openTelemetry, "quarkus-mongodb-client", spanNameExtractor)
+                .addAttributesExtractor(new CommandEventAttrExtractor())
+                .buildInstrumenter(SpanKindExtractor.alwaysClient());
+        LOGGER.debugf("MongoTracingCommandListener created");
+    }
+
+    @Override
+    public void commandStarted(CommandStartedEvent event) {
+        LOGGER.tracef("commandStarted event %s", event.getCommandName());
+
+        Context parentContext = Context.current();
+        if (instrumenter.shouldStart(parentContext, event)) {
+            Context context = instrumenter.start(parentContext, event);
+            requestMap.put(event.getRequestId(), new ContextEvent(context, event));
```

Comment:
I'm concerned with the amount of data you are storing here... can you give an example of one of this events?
Can you just store what you need later?

Reply:
@brunobat  it is not big, just description of mongo connection, db name and command. You can inspect it by using `BookResourceTest`. But I could create another record to just save command name and the command itself, if you prefer

Reply:
@brunobat I've [moved the implementation ](https://github.com/vkn/quarkus/tree/mongodb-otel-refactor) to opentelemetry extension. Is this enough for a new PR?

Reply:
It is but there are also other improvements needed:

> Also we lack documentation entries and additional IT tests for Uni, Multi, parent-child relationships and failure cases.
>
> Also, the assertions need to verify the attributes of the spans.

Reply:
@brunobat  Opened #40472

I'm not sure about additional IT tests. There is already a test for for reactive endpoint with assertion for attributes:
```
    @Test
    void reactiveClient() {
        testInsertBooks("/reactive-books");
        assertTraceAvailable("my-reactive-collection");
    }
```